x-pack/plugins/apm/public/hooks/useLocalStorage.ts

```diff
@@ -26,7 +24,9 @@ export function useLocalStorage<T>(key: string, defaultValue: T) {
     }
 
     return toStore;
-  }
+  }, [key, defaultValue]);
+
+  const [item, setItem] = useState<T>(getFromStorage());
```

Comment:
Can we avoid `useCallback` if the arguments are passed to `getFromStorage`?
``
```suggestion
  const [item, setItem] = useState<T>(getFromStorage(key, defaultValue));
```

Reply:
i got rid of the `useCallback`, but it still needs the `useEffect` with dependencies on `[key, defaultValue]` since changes in any of these should trigger a state change with results in a new value for the returned `item`. The lack of this update was the source of state state in this issue. In correlations `defaultValue` gets updates from a `useFetcher` response (for dynamic index pattern)